spec/exercises.js

```diff
@@ -44,18 +50,23 @@ describe('Testing exercises', function () {
   var exercises = exec.sync(['list', '--lang=en']).toString().split('\n')
   exercises.pop() // last line break
   exercises.map(function (exercise) {
-    return wUtil.idFromName(exercise)
-  }).forEach(function (id, nr) {
-    var folder = path.join(process.cwd(), 'test', id)
-    var allFiles = require('glob').sync('*.*', {
+    return {
+      name: exercise,
+      id: wUtil.idFromName(exercise, config.spaceChar)
+    }
+  }).forEach(function ({ name, id }, nr) {
+    var folder = path.join(process.cwd(), config.exercisesFolder || 'test', id)
+    var allFiles = require('glob').sync(config.files || '*.*', {
       cwd: folder
     })
 
     allFiles.filter(function (file) {
-      return /^(in)?valid(-|_)?(\d*)?\.\w+/.test(file)
+      var validRegex = config.validRegex || /^(in)?valid(-|_)?(\d*)?\.\w+/
+      var re = new RegExp(validRegex)
+      return re.test(file)
     }).forEach(function (file, fileNr) {
       it('./' + path.relative(process.cwd(), path.join(folder, file)) + ' (' + nr + ':' + fileNr + ')\t ', function (done) {
-        exec.async(['select', id], function (err, stdout, stderr) {
+        exec.async(['select', name], function (err, stdout, stderr) {
```

Comment:
The selection of the id was done on purpose to make sure that the selection is language independent and that id's can be selected. if you want to select by name then the `--lang=en` flag might be necessary.